### PR TITLE
TNR-2129: Add IdP Auth to connect API

### PR DESCRIPTION
**Pull request by bijan-vacasa**
https://vacasait.atlassian.net/browse/TNR-2129

Add IdP Auth to connect API

- Also cleaned up some retry logic (hopefully)

### CODE REVIEW

File: vacasa/connect/requests_config.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+class BackoffWithJitter:
+    """urllib3.util.retry.Retry expects a float for `backoff_factor`, but we want to sneak in some jitter """
+
+    def __init__(self, backoff, jitter_min, jitter_max):
+        self._backoff = backoff
+        self._jitter = (jitter_min, jitter_max)
+
+    def __mul__(self, other):
+        """Add some random jitter every time the backoff function tries to do some multiplication"""
+        return (self._backoff + random.uniform(*self._jitter)) * other
```
Comment (engstrom) [inline vacasa/connect/requests_config.py]: What do you think of this solution?  
https://github.com/abrookins/reliable_service_calls_with_python/blob/master/simulation/jittery_retry.py

**Comment by engstrom** [comment]
That's a lot of red. 🥇

File: vacasa/connect/connect.py [hunk 2]
```diff
@@ -1,40 +1,30 @@
...
+from .idp_auth import IdpAuth
+from .requests_config import request_with_retries
+from .util import log_http_error
 
 logger = logging.getLogger(__name__)
-
-
-def is_https_url(url: str) -> bool:
-    return urlparse(url).scheme.lower() == 'https'
+requests = request_with_retries()
```
Comment (CodyHelbling) [inline vacasa/connect/connect.py]: This is quite a bit cleaner. I like it.

File: vacasa/connect/requests_config.py (hunk 1 above)
Comment (bijan-vacasa) [reply vacasa/connect/requests_config.py]: That's way better, thanks!

**Pull request merged**